Public/Src/Cache/ContentStore/Interfaces/FileSystem/VfsUtilities.cs

```diff
@@ -0,0 +1,57 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.IO;
+using BuildXL.Cache.ContentStore.Hashing;
+using BuildXL.Cache.ContentStore.Interfaces.Sessions;
+using BuildXL.Cache.ContentStore.Interfaces.Utils;
+
+namespace BuildXL.Cache.ContentStore.Interfaces.FileSystem
+{
+    /// <summary>
+    /// Defines utility methods for working with VFS
+    /// </summary>
+    public static class VfsUtilities
+    {
+        private static readonly string DirectorySeparatorCharString = Path.DirectorySeparatorChar.ToString();
+        private static readonly char[] PathSplitChars = new[] { Path.DirectorySeparatorChar };
+        private static readonly char[] FilePlacementInfoFileNameSplitChars = new[] { '_' };
+
+        /// <summary>
+        /// Gets whether a path is contained in another path
+        /// </summary>
+        public static bool IsPathWithin(this string path, string candidateParent)
+        {
+            if (!path.StartsWith(candidateParent, StringComparison.OrdinalIgnoreCase))
+            {
+                return false;
+            }
+
+            if (candidateParent.EndsWith(DirectorySeparatorCharString))

```

Comment:
Why don't you just append `candidateParent` with `\` if it's not ending with `\`?

Reply:
This is check. No need to mutate the string.

---
In reply to: [298305804](https://github.com/microsoft/BuildXL/pull/519#discussion_r298305804) [](ancestors = 298305804)